pkg/masterapi/master_rest_api.go

```diff
@@ -0,0 +1,244 @@
+	fullUrl := "http://" +ip + ":" + port + endpoint
+	req, err := http.NewRequest("GET", fullUrl, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Content-type", "application/json")
+	if token != "" {
+		req.Header.Add("Authorization", "token="+token)
+	}
+	return req, nil
+}
+
+func executeAndValidateResponse(request *http.Request, logPrefix string) ([]byte, error) {
+	var byteContent []byte
+	var resp *http.Response
+
+	client := &http.Client{}
+	resp, err := client.Do(request)
+
+	if err != nil {
+		return nil, ErrorExecuteRequest(logPrefix, err)
+	}
+
+	defer resp.Body.Close()
+
+	if resp == nil {
+		return nil, ErrorEmptyResponse(logPrefix)
+	}
+
+	// Check response status code
+	if resp.Status == "" {
+		return nil, errors.New(logPrefix + " Empty response status\n")
+	}
+
+	byteContent, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf(logPrefix + " Error in ioutil.ReadAll: %s", err)
+	}
+	return byteContent, nil
+}
+
+func (mesosRestClient *GenericMasterAPIClient) createLoginRequest(endpoint string) (*http.Request, error) {
+	var jsonStr []byte
+	url := "http://" + mesosRestClient.MesosConf.MasterIP + endpoint //"/acs/api/v1/auth/login"
+
+	// Send user and password
+	jsonStr = []byte(`{"uid":"` + mesosRestClient.MesosConf.MasterUsername + `","password":"` + mesosRestClient.MesosConf.MasterPassword + `"}`)
+
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	return req, nil
+}
+
+func (mesosRestClient *GenericMasterAPIClient) getDebugModeState() ([]byte, error) {
+	fmt.Println("========= getDebugModeState() : DEBUG MODE =============")
+	filePath, exists := mesosRestClient.DebugProps["file"]
+	if !exists {
+		return nil, fmt.Errorf("Missing 'file' parameter in debug props:  %s", mesosRestClient.DebugProps)
+	}
+	byteContent, e := ioutil.ReadFile(filePath)
+	if e != nil {
+		return nil, fmt.Errorf("File error:  %s", e)
+	}
+	return byteContent, nil
+}
+
+// ========================================= State Request Parser ===================================================
+
+type GenericMasterStateParser struct {
+	//Message *conf.MesosState
+	Message *data.MesosAPIResponse
+}
+
+const GenericMasterStateParserClass = "[GenericMasterStateParser]"
+
+func (parser *GenericMasterStateParser) parseResponse(resp []byte) error {
+	glog.V(3).Infof(GenericMasterStateParserClass + " in parseAPIStateResponse")
+	if resp == nil {
+		return ErrorEmptyResponse(GenericMasterStateParserClass)
+	}
+
+	var jsonMesosMaster data.MesosAPIResponse //conf.MesosState
+	err := json.Unmarshal(resp, &jsonMesosMaster)
+	if err != nil {
+		return fmt.Errorf(GenericMasterStateParserClass + " Error in json unmarshal for state response : %s", err)
+	}
+	parser.Message = &jsonMesosMaster
+	return nil
+}
+
+func (parser *GenericMasterStateParser) GetMessage() interface{} {
```

Comment:
why not just let parser to return the message, something like:
```go
msg, err := genericMasterStateParser.parseResponse(resp)
```
The current implementation requires you to call two methods